api/credentials/internal/identity.go

```diff
@@ -91,6 +93,26 @@ func orMatcher(list []IdentityMatcher) IdentityMatcher {
 // ConsumerIdentity describes the identity of a credential consumer.
 type ConsumerIdentity map[string]string
 
+func (c *ConsumerIdentity) UnmarshalJSON(data []byte) error {
+	m := map[string]interface{}{}
+	err := runtime.DefaultJSONEncoding.Unmarshal(data, &m)
+	if err != nil {
```

Comment:
should be 
```
var m  map[string]interface{}
```
without initialization

Reply:
why?

Reply:
Depends on the implementation. For example your unmarshaller always creates a new object. I would not rely on this, to get a clean result,